aten/src/ATen/native/TensorShape.h

```diff
@@ -51,6 +51,38 @@ inline int64_t get_num_splits(const Tensor& self, int64_t split_size, int64_t di
   return num_splits;
 }
 
+inline std::tuple<int64_t, int64_t> get_slice_range(
+    c10::optional<int64_t> start,
+    c10::optional<int64_t> end,
+    const int64_t dim_size) {
+
+  int64_t start_val = start.has_value() ? start.value() : 0;
+  int64_t end_val = end.has_value() ? end.value() : INT64_MAX;
+
+  // INT64_MAX stands for default value.
+  if (start_val == INT64_MAX) {
```

Comment:
rando question is this essentially maybe_wrap_dim for start and end?

Reply:
It's definitely clamp with start clamped to [0, dim_size] and end clamped to [start, dim_size]. Probably should be rewritten cleaner at some point :)